doc/asciidoc/configuration.adoc

```diff
@@ -32,11 +32,56 @@ containing the above code is `file.sail`.
 sail --ocaml --config file.json file.sail
 ----
 
+=== JSON representation of Sail types
+
+The following example demonstrates how basic Sail types are
+represented in the JSON configuration file. We can load integers,
+booleans, and strings, all of which correspond directly to the
+equivalent JSON type. Integers are allowed to be arbitrary precision,
+as they are in Sail.
+
+:confbasic: sail_doc/config_basic_types.json
+
+sail::example[from=confbasic,part=body,dedent]
+
+The following JSON can be used to instantiate the above Sail source:
+
+[source,json]
+----
+include::sail_config/config_basic_types.json[]
+----
+
+Note the two permissible representations for bitvectors. First, we can
+represent them as a list of JSON booleans. This representation has the
+advantage of being simple and unambiguous regarding the length of the
+bitvector, but is verbose. Second, we can use a string containing the
+bitvector value formatted as a Sail bitvector literal (so `pass:[_]`
+can be used as a separator) with an explicit integer length. If the
+value is shorter than the desired width it will be zero-extended. If
+it is larger then any high bits above the stated width will be ignored
+(care should be taken to avoid this). In addition to `0b` and `0x`
```

Comment:
Shouldn't larger value be an error instead of silent truncation? Or do you have a use-case where we need this?

Reply:
Yes, it might be more sensible to just error here. Not sure why I didn't just make it do that when I was implementing it.

I think maybe I was thinking that anything accepted by the generated schema should load without error, and this over-long bitvector case is the one thing I can't really detect that way.

Reply:
Also I don't want to delay this getting merged if it is annoying to implemented :) Would be great to see this land soon.